### ISSUE TITLE
SOCI_STATIC and plugins

### ISSUE THREAD

**Issue by wisk** [SOCI/soci#497]
Hi there,

I've a question regarding the CMake variable ``SOCI_STATIC``.
My wish is to build SOCI in static only (no .so/.dll dependencies) and use the backend **SQLite3**, to do so I compiled SOCI using the CMake switch ``SOCI_STATIC``.
To link my program; I used both **libsoci_core_4_0.{lib,so}** and **libsoci_sqlite3_4_0.{lib,so}**.
However, when the program tries to call session constructor, it throws an error **Failed to find shared library for backend sqlite3**.
Using procmon, I figured out that the SOCI core tried to load **soci_sqlite3_4_0.dll**, so I copied it the current folder. But since this DLL depends on **soci_core_4_0.dll**, I also had to copy this library to the current library in order to use SOCI.
So I'm wondering if it's possible to only link with static libraries and avoid SOCI to load shared library dynamically.

Thanks in advance¸

**Comment by mloskot** [comment]
On 22 August 2016 at 23:08, wisk <notifications@github.com> wrote:

> Hi there,
>
> I've a question regarding the CMake variable SOCI_STATIC.
> My wish is to build SOCI in static only (no .so/.dll dependencies) and use
> the backend *SQLite3*, to do so I compiled SOCI using the CMake switch
> SOCI_STATIC.
> To link my program; I used both *libsoci_core_4_0.{lib,so}* and
> *libsoci_sqlite3_4_0.{lib,so}*.
>
You should be able to link against static only libraries:
libsoci_core_4_0.lib and libsoci_sqlite3_4_0.lib.

I suppose, you might be mixing linking against static libraries and import
libraries (for DLLs).
Try to delete the build and re-build SOCI with statiic-only binaries:
cmake    -DSOCI_SHARED=OFF    -DSOCI_STATIC=ON ......

Regards,
Mateusz Loskot

**Comment by wisk** [comment]
Thanks for your quick answer. :)

I should have told you I'm using hunter to handle SOCI as a dependency, but it shouldn't change anything.
I tried to set SOCI_SHARED=OFF and SOCI_STATIC=ON, accordingly to this configuration I only have lib<libname>.lib generated.
The CMake script call the following command ``target_link_libraries(db_soci medusa ${SOCI_SQLITE3_PLUGIN} ${SOCI_LIBRARY})``.
``SOCI_SQLITE3_PLUGIN`` points to ``libsoci_sqlite3_4_0.lib``.
``SOCI_LIBRARY`` points to ``libsoci_core_4_0.lib``.
And yet I still have the same issue.

...And I get it, I made several errors here, I'm detailing it because it might help someone else:
- My first mistake was to assume that ``session("sqlite3", ...)`` will try to find the module statically. Actually to do so, it's required to call ``session(soci::sqlite3, ..)`` instead.
- My second mistake was to used a slightly updated version of ``FindSoci.cmake`` presents in the repository (, btw I plan to provide an updated version).
I used ``SOCI_SQLITE3_PLUGIN`` variable instead of ``SOCI_sqlite3_PLUGIN``.
- Last but not least, I still need to link my project against ``sqlite3.lib``.

Sorry for the noise.

Best regards,
